frontend/src/concepts/pipelines/content/experiment/CreateExperimentButton.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { Button } from '@patternfly/react-core';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { ExperimentKFv2 } from '~/concepts/pipelines/kfTypes';
+import ManageExperimentModal from '~/concepts/pipelines/content/experiment/ManageExperimentModal';
+
+type CreateExperimentButtonProps = {
+  onCreate?: (experiment: ExperimentKFv2) => void;
+} & Omit<React.ComponentProps<typeof Button>, 'onClick'>;
+
+const CreateExperimentButton: React.FC<CreateExperimentButtonProps> = ({
+  onCreate,
+  children,
+  ...buttonProps
+}) => {
+  const { apiAvailable, refreshAllAPI, pipelinesServer } = usePipelinesAPI();
+  const [open, setOpen] = React.useState(false);
+
+  return (
+    <>
+      <Button
+        {...buttonProps}
+        isDisabled={!apiAvailable || buttonProps.isDisabled || !pipelinesServer.compatible}
+        onClick={() => setOpen(true)}
+      >
+        {children || 'Create experiment'}
+      </Button>
+      <ManageExperimentModal
+        isOpen={open}
+        onClose={(experiment) => {
+          setOpen(false);
+          if (experiment) {
+            if (onCreate) {
+              onCreate(experiment);
+            }
```

Comment:
nit: could simplify to `experiment && onCreate?.(experiment)`

Reply:
There is still a `refreshAllAPI();` function to be called below.

Reply:
Ok, then the nit is just simplifying the condition for onCreate.

Reply:
the eslint is preventing me from doing that `onCreate && onCreate(experiment);`
`Expected an assignment or function call and instead saw an expression.`

Reply:
As shown in the first comment, this is what my intention was `onCreate?.(experiment)` but this is a nitpick and I don't want us to waste too much time on this, so no worries.